cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -546,25 +555,26 @@ runTxBuild
         & onLeft (left . ShelleyTxCmdQueryConvenienceError . AcqFailure)
         & onLeft (left . ShelleyTxCmdQueryConvenienceError . QceUnsupportedNtcVersion)
 
+      Refl <- testEquality era nodeEra
+              & hoistMaybe (ShelleyTxCmdTxEraCastErr $ EraCastError ("nodeEra" :: Text) era nodeEra)
```

Comment:
We need to update the error rendering of `ShelleyTxCmdTxEraCastErr` to basically say the cli must produce transactions that are of the same era as the node. This can be done in a follow up PR.